contrib/python-bindings/include/cell_accessor_wrapper.h

```diff
@@ -193,22 +193,22 @@ namespace python
   };
 
 
-  template <int dim, int spacedim>
+  template <int dim_, int spacedim_>
   const CellAccessorWrapper
   CellAccessorWrapper::construct_neighbor_wrapper(const int i) const
   {
-    CellAccessor<dim, spacedim> *cell =
-      static_cast<CellAccessor<dim, spacedim> *>(cell_accessor);
+    Assert(dim_ == dim && spacedim_ == spacedim, ExcInternalError());
```

Comment:
You could make this a `static_assert`.

Reply:
No, `dim` and `spacedim` are member variables. Apart from shadowing, this is the source of the compile errors in the first place.

Reply:
Oh yes, of course!